src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/ConnectionHandlerContext.cs

```diff
@@ -31,5 +35,51 @@ internal class ConnectionHandlerContext : HandlerRequest
         /// Class used by orchestrator while chaining handlers.
         /// </summary>
         public Exception Error { get; set; }
+
+        /// <summary>
+        /// The Guid of the Connection.
+        /// </summary>
+        public Guid ConnectionId { get; internal set; } = Guid.Empty;
+
+        /// <summary>
+        /// The SslStream used by the connection.
+        /// </summary>
+        public SslStream SslStream { get; internal set; }
+
+        /// <summary>
+        /// The SslOverTdsStream used by the connection in case of Tds below 7.4.
+        /// </summary>
+        public SslOverTdsStream SslOverTdsStream { get; internal set; }
```

Comment:
This ties to my previous point about blocking SQL Server versions. TDS < 7.4 is SQL Server 2008 R2 and below, and the [SqlClient documentation](https://learn.microsoft.com/en-us/sql/connect/ado-net/sqlclient-driver-support-lifecycle?view=sql-server-ver16#sql-version-compatibility-with-microsoftdatasqlclient) is pretty specific that v5.2 and above only supports SQL Server 2014 and above (TDS 7.4.)

If we want to comply with the documentation, we can remove `SslOverTdsStream`. If we want to remain consistent with .NET Core and keep support for SQL Server 2005+, we need to keep it.

Personally: SQL Server 2008 R2 went out of support in July 2019. It's not documented as supported as far as I know. I'd vote to harmonise the minimum supported version for every type of SNI across .NET Core & .NET Framework at TD 7.4 / SQL Server 2012, and throw an exception for anything earlier.

This also has implications for any ported code which refers to the `Is2000`, `Is2005OrNewer` and the `Is2008OrNewer` properties on `SqlConnection` and `SqlInternalConnection`.

Reply:
Ya I agree. My intention was to get the code as is, and get it to compile with Handlers. 

@cheenamalhotra  has also pointed out removing code for old servers, (out of support ones). I will tackle this in subsequent commits.

Reply:
@edwardneal 
>If we want to comply with the documentation, we can remove SslOverTdsStream. If we want to remain consistent with .NET Core and keep support for SQL Server 2005+, we need to keep it.

`SslOverTdsStream` will be used as long as TDS 7.4 is available for use. It is still a widely used protocol and co-exists with TDS8 in Azure SQL DB as well.

Reply:
Having just re-checked the TDS spec, you're completely right sorry - TDS 7.4 is used up to SQL Server 2019, so we're a long way off removing this.